uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Helper/ContactActionsMenu/WorkflowActionGroup.php

```diff
@@ -0,0 +1,283 @@
+
+    if ($lineManagers) {
+      $lineManagersListItem = new LineManagersListItem($this->managerService, $this->contactID);
+      $actionsGroup->addItem($lineManagersListItem);
+      $actionsGroup->addItem($this->getManageLineManagerButton());
+    }
+    else {
+      $noLineManagerTextItem = new NoSelectedLineManagerTextItem();
+      $actionsGroup->addItem($noLineManagerTextItem);
+      $actionsGroup->addItem($this->getAddLineManagerButton());
+    }
+
+    return $actionsGroup;
+  }
+
+  /**
+   * Gets the Workflow Joining button
+   *
+   * @return ActionsGroupButtonItem
+   */
+  private function getWorkflowJoiningButton() {
+    $caseTypeId = $this->getCaseTypeID('Joining');
+    $url = CRM_Utils_System::url(
+      'civicrm/tasksassignments/dashboard#/tasks',
+      "reset=1&cid=$this->contactID&openModal=assignment&caseTypeId=$caseTypeId"
+    );
+    $params = [
+      'label' => 'Joining',
+      'class' => 'btn btn-primary-outline',
+      'icon' => 'fa fa-user-plus',
+      'url' => $url
+    ];
+    return $this->getMenuButton($params);
+  }
+
+  /**
+   * Gets the Workflow Exiting button
+   *
+   * @return ActionsGroupButtonItem
+   */
+  private function getWorkflowExitingButton() {
+    $caseTypeId = $this->getCaseTypeID('Exiting');
+    $url = CRM_Utils_System::url(
+      'civicrm/tasksassignments/dashboard#/tasks',
+      "reset=1&cid=$this->contactID&openModal=assignment&caseTypeId=$caseTypeId"
+    );
+    $params = [
+      'label' => 'Exiting',
+      'class' => 'btn btn-primary-outline',
+      'icon' => 'fa fa-user-times',
+      'url' => $url
+    ];
+
+    return $this->getMenuButton($params);
+  }
+
+  /**
+   * Gets the Workflow Other button
+   *
+   * @return ActionsGroupButtonItem
+   */
+  private function getWorkflowOtherButton() {
+    $url = CRM_Utils_System::url(
+      'civicrm/tasksassignments/dashboard#/tasks',
+      "reset=1&cid=$this->contactID&openModal=assignment"
+    );
+    $params = [
+      'label' => 'Other...',
+      'class' => 'btn btn-primary-outline',
+      'icon' => '',
+      'url' => $url
+    ];
+
+    return $this->getMenuButton($params);
+  }
+
+  /**
+   * Gets the Workflow New Task button
+   *
+   * @return ActionsGroupButtonItem
+   */
+  private function getWorkflowNewTaskButton() {
+    $url = CRM_Utils_System::url(
+      'civicrm/tasksassignments/dashboard#/tasks',
+      "reset=1&cid=$this->contactID&openModal=task"
+    );
+    $params = [
+      'label' => 'New Task',
+      'class' => 'btn btn-primary-outline',
+      'icon' => 'fa fa-check-square-o',
+      'url' => $url
+    ];
+
+    return $this->getMenuButton($params);
+  }
+
+  /**
+   * Gets the Workflow New Document button
+   *
+   * @return ActionsGroupButtonItem
+   */
+  private function getWorkflowNewDocumentButton() {
+    $url = CRM_Utils_System::url(
+      'civicrm/tasksassignments/dashboard#/documents',
+      "reset=1&cid=$this->contactID&openModal=document"
+    );
```

Comment:
This URL is repeated in all of the buttons with some small variations. Do you think you can extract it to a separate method?